pyvrp/cpp/Solution.h

```diff
@@ -76,21 +81,41 @@ class Solution
     Duration timeWarp_ = 0;       // Total time warp over all routes
 
     Routes routes_;  // Routes - only includes non-empty routes
+    std::vector<Routes> routesPerVehicleType_;  // Routes per vehicle type
     std::vector<std::pair<Client, Client>> neighbours;  // pairs of [pred, succ]
+    std::vector<RouteType> assignedVehicleTypes;  // Client assigned veh. types
```

Comment:
I removed `routesPerVehicleType_` as I'm not currently using it indeed.

I think `assignedVehicleTypes ` makes sense just like `neighbours` (we could also compute that on the fly). This will be checked repeatedly in `operator==` (used in population). It's as expensive as computing `neighbours` (or less), which I think is negligible overall.

Also, we should soon discuss a distance measure that takes into account vehicle assignments (in a seperate PR) which would use this (just like `brokenPairsDistance` uses `neighbours`).